app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/MosaicImplicits.scala

```diff
@@ -145,40 +147,56 @@ class MosaicImplicits[HistStore: HistogramStore](mtr: MetricsRegistrator,
             // Assume that we're in a single band case. It isn't obvious what it would
             // mean if the band count weren't 3 or 1, so just make the assumption that we
             // wouldn't do that to ourselves and don't handle the remainder
-            BacksplashMosaic.getStoreHistogram(filtered, histStore) flatMap {
-              hists =>
-                {
-                  val ioMBT = (BacksplashMosaic.filterRelevant(self) map {
-                    relevant =>
-                      println("yup in the thing")
-                      val time = readSceneTmsTimer.time()
-                      val img = relevant.read(z, x, y) map { im =>
-                        ColorRampMosaic.colorTile(
-                          im,
-                          hists,
-                          relevant.singleBandOptions getOrElse {
-                            throw SingleBandOptionsException(
-                              "Must specify single band options when requesting single band visualization.")
-                          }
-                        )
-                      }
-                      println(s"Image bands: ${img map { _.bandCount }}")
-                      time.stop()
-                      img
-                  }).collect({ case Some(mbtile) => mbtile }).compile.toList
-                  mergeTiles(ioMBT).map {
-                    case Some(t) => Raster(t, extent)
-                    case _       => Raster(ndtile, extent)
-                  }
-                }
+            val ioMBTwithSBO = (BacksplashMosaic.filterRelevant(self) map {
+              relevant =>
+                println("yup in the thing")
+                logger.debug(s"Band Subset Required: ${relevant.subsetBands}")
+                val img = relevant.read(z, x, y)
+                img.map(i =>
+                  logger.debug(s"Band Count Available: ${i.bandCount}"))
+                (img, relevant.singleBandOptions)
+            }).collect({ case (Some(mbtile), Some(sbo)) => (mbtile, sbo) })
+              .compile
+              .toList
+
+            for {
+              histograms <- histStore.projectHistogram(
+                UUID.fromString("9fc60cbb-17bd-4376-b7dd-0c8b2ba7383b"),
```

Comment:
Should be project ID - but that's not available anywhere here

Reply:
Now available on images